docs/adr/0039-use-react-container-component-abstraction.md

```diff
@@ -0,0 +1,93 @@
+
+Pros and cons:
+
+- Good, because it's better inversion of control than what we have now while keeping a similar pattern
+- Good, because it allows us to move away from the `useAccount` hook and makes it easier to see what data needs to be fetched and used per page
+- Bad, because it's preferable to use hooks over `apolloClient` direct methods
+- Bad, because `apolloClient` and `authClient` should be singletons that we would need to pass into every new page model, or alternatively always pull these out from Context
+- Bad, because our loading and error states may still be inconsistent without more refactoring
+
+### Per-page container components and models, validate in container
+
+[See the prototype](https://github.com/mozilla/fxa/pull/15493/) for this option that helped informed the decision outcome.
+
+Description:
+
+- Every page component that needs an API call or data validation will have a corresponding container component to handle them
+- Wrap the application in `<ApolloProvider>` so the container component can execute queries and mutations with hooks
+- Only perform API calls in container components. This includes `useQuery` for page-specific needed data, which will be stored in and then read from Apollo Cache in future queries, `useMutation` mutations, and `authClient` calls (until those can be refactored to use GQL)
+- Handle all error processing in the container component
+- If the page needs any validation, create a separate model containing the data to validate, and perform the validation in the container component
+
+Pros and cons:
+
+- Good, because container components are a common pattern and this approach provides a better separation of concerns among our API/network/business logic layer, data models, and presentation layer
+- Good, because of more flexibility and inversion of control. We won’t have to mock entire models or partials with `as unknown as Account` type casting and can instead, mock only the handlers and their results when writing page-level tests
+- Good, because it allows us to move away from the `useAccount` hook and makes it easier to see what data needs to be fetched and used per page
+- Good, because placing validation and API handlers in the same location is explicit and clear
+- Good, because of benefits gained when using Apollo Client's hooks (see "More Info")
+- Neutral, because GQL related container tests will require a `MockedProvider`. Most tests will be presentation/page-level and will _not_ require `MockedProvider`, but while this is standard when testing Apollo Client and will give us a higher degree of confidence in our API layer, it's another testing tool that engineers will need to familiarize themselves with
+- Neutral, because we can isolate changes to non-Settings pages first without taking on a massive refactoring effort, but differing patterns between the two may feel awkward
```

Comment:
I can probably add a con here about the initial refactoring effort, but we can do this when we're working on the reset password redesign which would require refactoring regardless, so 🤔